src/meta.rs

```diff
@@ -2,21 +2,78 @@ use crate::id::ResourceId;
 
 use anyhow::Error;
 use canonical_path::CanonicalPathBuf;
+use serde::{Deserialize, Serialize};
+use std::any;
 use std::ffi::{OsStr, OsString};
+use std::fs::{self, File};
+use std::io::Write;
+use std::path::Path;
 use std::time::SystemTime;
 use walkdir::DirEntry;
 
-#[derive(Eq, PartialEq, Hash, Clone, Debug)]
+const METADATA_RELATIVE_PATH: &str = ".ark/meta";
+
+#[derive(Eq, PartialEq, Hash, Clone, Debug, Deserialize, Serialize)]
 pub struct ResourceMeta {
     pub id: ResourceId,
-    pub modified: SystemTime,
+    // pub modified: SystemTime,
+    pub created_date: SystemTime,
```

Comment:
We need `modified` field for tracking when it's necessary to re-scan a resource. It's better to put `created_date` into `extra` for now.